File: app/src/main/java/com/jorotayo/algorubickrevamped/ObjectBox.java
package com.jorotayo.algorubickrevamped;

import android.app.PendingIntent;
import android.content.Context;
import android.util.Log;

import com.jorotayo.algorubickrevamped.data.MyObjectBox;

import io.objectbox.BoxStore;
import io.objectbox.android.AndroidObjectBrowser;

public class ObjectBox {
    private static BoxStore boxStore;

    public static void init(Context context) {
        if (boxStore == null) {
            boxStore = MyObjectBox.builder().androidContext(context.getApplicationContext()).build();
        }

        if (BuildConfig.DEBUG) {
            new AndroidObjectBrowser(boxStore).start(context.getApplicationContext());
        }
    }

    public static BoxStore getBoxStore() {
        return boxStore;
    }
}

File: app/src/main/java/com/jorotayo/algorubickrevamped/ObjectBox.kt
package com.jorotayo.algorubickrevamped

import android.content.Context
import com.jorotayo.algorubickrevamped.data.MyObjectBox
import io.objectbox.BoxStore
import io.objectbox.android.AndroidObjectBrowser

object ObjectBox {
    @Volatile
    private var boxStore: BoxStore? = null

    @JvmStatic
    fun init(context: Context) {
        if (boxStore == null) {
            synchronized(this) {
                if (boxStore == null) {
                    boxStore = MyObjectBox.builder()
                        .androidContext(context.applicationContext)
                        .build()
                }
            }
        }
        if (BuildConfig.DEBUG) {
            AndroidObjectBrowser(boxStore).start(context.applicationContext)
        }
    }

    @JvmStatic
    fun getBoxStore(): BoxStore =
        boxStore ?: error("ObjectBox not initialized. Call ObjectBox.init() first.")
}
